addons/mail/models/mail_channel.py

```diff
@@ -379,8 +379,8 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
                 'members': [('insert', members_data)],
             })
         if invite_to_rtc_call:
-            if self.env.user._is_public() and 'guest' in self.env.context:
-                guest = self.env.context.get('guest')
+            guest = self.env['mail.guest']._get_guest_from_context()
+            if self.env.user._is_public() and guest:
```

Comment:
Maybe not for this PR, but this pattern seems frequent enough to deserve its dedicated method.